src/Saml2/Provider.php

```diff
@@ -110,28 +170,64 @@ public function redirect()
     {
         $this->request->session()->put('state', $state = $this->getState());
```

Comment:
```suggestion
        $state = null;

        if ($this->usesState()) {
            $this->request->session()->put('state', $state = $this->getState());
        }
```

Reply:
Added a slightly different code so the `null` state is not appended at all.